Packs/Mimecast/Integrations/MimecastEventCollector/MimecastEventCollector.py

```diff
@@ -0,0 +1,567 @@
+        post_body['data'][0]['fileFormat'] = 'json'
+        if self.token:
+            post_body['data'][0]['token'] = self.token
+
+        return json.dumps(post_body)
+
+    def write_file(self, file_name: str, data_to_write: bytes):
+        """
+        Args:
+            - file_name (str): The name of the file returned form the api response header
+            - data_to_write (bytes): The byte's representation of the zip file.
+
+        Returns:
+             The events that were stored in the zip-file files.
+        """
+        if '.zip' in file_name:
+            try:
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    byte_content = io.BytesIO(data_to_write)
+                    zip_file = ZipFile(byte_content)
+                    zip_file.extractall(tmpdir)
+                    extracted_logs_list = []
+                    for file in os.listdir(tmpdir):
+                        with open(os.path.join(tmpdir, file)) as json_res:
+                            extracted_logs_list.extend(self.process_siem_events(json.load(json_res)))
+
+                    return extracted_logs_list
+
+            except Exception as e:
+                return_error('Error writing file ' + file_name + '. Cannot continue. Exception: ' + str(e))
+
+        else:
+            return_error('We support only compressed siem logs request.')
+
+
+class MimecastGetAuditEvents(IntegrationGetEvents):
+
+    def __init__(self, client: MimecastClient, options: IntegrationOptions):  # pragma: no cover
+        super().__init__(client=client, options=options)
+        self.page_token: str = ''
+        self.start_time: str = ''
+        self.end_time: str = self.to_audit_time_format(
+            datetime.now().astimezone().replace(microsecond=0).isoformat())
+        self.uri: str = '/api/audit/get-audit-events'
+
+    @staticmethod
+    def get_last_run(events: list) -> dict:
+        pass
+
+    def run(self):
+        self.options.limit = 1 if demisto.command() == 'test-module' else None
+        return super().run()
+
+    def _iter_events(self):
+        self.client.request = IntegrationHTTPRequest(**(self.get_req_object_audit()))
+        response = self.call()
+        events = self.process_audit_response(json.loads(response.text))
+        if not events:
+            return []
+
+        while True:
+            demisto.info(f'\n {len(events)} Audit logs were fetched from mimecast \n')
+            yield events
+
+            self.client.request = IntegrationHTTPRequest(**(self.get_req_object_audit()))
+            response = self.call()
+            events = self.process_audit_response(json.loads(response.text))
+            if not events:
+                break
+
+    def process_audit_response(self, res: dict):
+        """
+        Args:
+            res (dict) - This method gets the response.text from the get Audit events.
+        Returns:
+            event_list (list) - The processed audit events
```

Comment:
In the `res` docstring Audit is with capital A. Decide on standard naming.

Reply:
Done